docs/src/extend/custom-processors.md

```diff
@@ -139,6 +139,16 @@ See [Specify a Processor](../use/configure/plugins#specify-a-processor) in the P
 
 ## File Extension-named Processor
 
+::: warning
+With the introduction of [flat configuration](../use/configure/configuration-files-new.md), this feature is being removed. If you are updating your plugin to provide flat configs, you **must** specify all processors explicitly in those configs.
+
+In ESLint v8.0.0+, if a user is still using the old configuration style, processors with file extension names will continue to be automatically included. Users who have switched to using flat configuration may need to manually add these processors to their configuration.
+
+**From ESLint v9.0.0+, when flat configuration becomes required, this feature will be completely removed.**
+
+This feature will continue to be available via `FlatCompat` in [`@eslint/eslintrc`](https://www.npmjs.com/package/@eslint/eslintrc), but there is no intention to continue support for it.
```

Comment:
I think we can shorten this to remove confusion and ensure people read it. :)

```suggestion
This feature is deprecated and only works in eslintrc-style configuration files. Flat config files do not automatically apply processors; you need to explicitly set the `processor` property.
```

Reply:
Sure, I'll update the other warning in the plugin docs as well to reflect the simplified language.

Is it worth explicitly noting that the feature will be removed in ESLint v9? I was trying to make plugin authors very aware that this change is coming, and if they get bug reports from users switching to v9, they have an easy to spot "oh that's why" reference in the docs.